src/test/ui/asm/type-check-1.rs

```diff
@@ -59,6 +61,11 @@ fn main() {
     }
 }
 
+unsafe fn generic<T>() {
+    asm!("{}", sym generic::<T>);
```

Comment:
This fails with `error: unconstrained generic constant` and I'm not exactly sure how to go about fixing it.

Reply:
To be fair, I feel like this should work. I don't consider it a blocker for landing this PR, but I suspect it may require some significant changes to the approach (again) in order to make it work.